airflow/providers/amazon/aws/hooks/logs.py

```diff
@@ -96,7 +104,15 @@ def get_log_events(
 
             yield from events
 
-            if next_token != response["nextForwardToken"]:
+            if "nextForwardToken" not in response:
+                num_consecutive_empty_response += 1
```

Comment:
Looking at the code before, we were always having a `nextForwardToken` in the response (otherwise we would have been getting a key error exceptions). So I think this will rarely be true. I think what we should consider "empty" is a response with zero events, so I'd use the condition of `if not event_count: ...` to incr `num_consecutive_empty_response` instead. WDYT?

Reply:
That was the condition we were using before #20814. [See here](https://github.com/apache/airflow/pull/20814/files#diff-3a03bd96408e5044757e85307f1fd83e7e86d2723212c6b36d34e6f500e4790bL102).

Reply:
I actually missed this too the first time I looked at it. But if you look at the top of that diff you linked the while loop statement actually changed as well. It used to be:
```
while event_count > 0:
```
So it was the while loop breaking when event_count was zero in most all cases, not the token (because again the token is always present basically).

Reply:
Ohhhhhh! Nice catch! :D I'll update the PR accordingly :) Thank you!

Reply:
I updated the PR. Let me know what you think :)

Reply:
Marking the conversation as resolved, feel free to reopen it if any concerns